streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobScheduler.scala

```diff
@@ -177,9 +177,12 @@ class JobScheduler(val ssc: StreamingContext) extends Logging {
             jobSet.totalDelay / 1000.0, jobSet.time.toString,
             jobSet.processingDelay / 1000.0
           ))
-          listenerBus.post(StreamingListenerBatchCompleted(jobSet.toBatchInfo))
+          listenerBus.post(StreamingListenerBatchCompleted(jobSet.toBatchInfo()))
         }
       case Failure(e) =>
+        val jobSet = jobSets.get(job.time)
+        val errorMessage = Some(Utils.exceptionString(e))
+        listenerBus.post(StreamingListenerBatchCompleted(jobSet.toBatchInfo(errorMessage)))
```

Comment:
The jobset may not have completed at this point! There may be multiple streaming jobs in the jobset, and when one streaming job fail, other streaming job may pass. So even if there is a failure, you have to keep other "jobs" in the batch that can finish successfully. 

So the right thing to do is to keep track of the failures in the JobSet, at the granularity of jobs. Eventually we may want to show the details failure information - e.g. 1 out of 3 output operations failed in Spark Streaming